src/plugins/map/MQ2MapCommands.cpp

```diff
@@ -666,15 +669,17 @@ void MapSetLocationCmd(SPAWNINFO* pChar, char* szLine)
 
 		MakeMapLoc(params, label, tag, pos, isDefaultLocSettings);
 
+		std::string labelStr;
+		if (label[0] == 0)
+			labelStr = sMapLocs[sMapLocs.size() - 1]->GetLabelText();
```

Comment:
does this protect against sMapLocs.size() == 0?

Reply:
not protected per se, but it should always be safe.  Line 670 has MakeMapLoc which will guarantee a new loc existing in the array (so minimum size of 1).